File: src/test/java/org/zbus/examples/pubsub/Sub.java
package org.zbus.examples.pubsub;

import java.io.IOException;

import org.zbus.broker.Broker;
import org.zbus.broker.BrokerConfig;
import org.zbus.broker.SingleBroker;
import org.zbus.mq.Consumer;
import org.zbus.mq.Consumer.ConsumerHandler;
import org.zbus.mq.Protocol.MqMode;
import org.zbus.net.http.Message;

public class Sub {
	@SuppressWarnings("resource")
	public static void main(String[] args) throws Exception{  
		//1）创建Broker代表
		BrokerConfig config = new BrokerConfig();
		config.setBrokerAddress("127.0.0.1:15555"); 
		final Broker broker = new SingleBroker(config);
		
		//2) 创建消费者 
		Consumer c = new Consumer(broker, "MyPubSub", MqMode.PubSub); 
		final String topic = "sse";
		c.setTopic(topic); 
		
		c.onMessage(new ConsumerHandler() { 
			int count = 0;
			@Override
			public void handle(Message msg, Consumer consumer) throws IOException {  
				if(topic.equals(msg.getTopic())){
					count++;
					System.out.println(topic+":" + count);
				} else {
					System.out.println(msg);
				}
			}
		});
		
		c.start(); 
	} 
}

File: api/csharp/examples/PubSub/Sub.cs
using Zbus.Net;
using Zbus.Mq;
using Zbus.Broker;

namespace Zbus.Examples.PubSub
{

    class MyHandler : IMessageHandler
    {
        public void Handle(Message msg, Consumer consumer)
        {
            System.Console.WriteLine(msg);
        }
    }

    class Sub
    {
        public static void Main(string[] args)
        {  
            IBroker broker = new SingleBroker(); //using BrokerConfig to change default

            Consumer c = new Consumer(broker, "MyPubSub", MqMode.PubSub);
            c.ConsumeTimeout = 30000;
            c.Topic = "zbus";

            c.OnMessage(new MyHandler());
            c.Start();
            //c.Stop();

            //broker.Dispose();
        }
    }
}
